lib/spack/spack/spec.py

```diff
@@ -1428,9 +1553,15 @@ def satisfies(self, other, deps=True, strict=False):
         elif strict and (other.architecture and not self.architecture):
             return False
 
+        if not self.compiler_flags.satisfies(other.compiler_flags, strict=strict):
+            return False
+
         # If we need to descend into dependencies, do it, otherwise we're done.
         if deps:
-            return self.satisfies_dependencies(other, strict=strict)
+            deps_strict = strict
+            if not (self.name and other.name):
+                deps_strict=True
```

Comment:
@becker33 Do you remember why this was added here?

Reply:
To explain the context, I'm trying to get a more formally correct implementation of `satisfies` while fixing #35597 see #35599